packages/common/src/feedback/fetch.ts

```diff
@@ -0,0 +1,69 @@
+import { IItem, getItem } from "@esri/arcgis-rest-portal";
+import {
+  IHubFeedback,
+  IHubRequestOptions,
+  IModel,
+  fetchModelFromItem,
+  fetchSetting,
+  getDefaultEntitySettings,
+  getItemBySlug,
+  isGuid,
+} from "..";
+import { PropertyMapper } from "../core/_internal/PropertyMapper";
+import { getPropertyMap } from "./_internal/getPropertyMap";
+import { computeProps } from "./_internal/computeProps";
+
+/**
+ * @private
+ * Get a Feedback entity by id or slug
+ * @param identifier item id or slug
+ * @param requestOptions request options
+ * @returns a promise that resolves a IHubFeedback object
+ */
+export async function fetchFeedback(
+  identifier: string,
+  requestOptions: IHubRequestOptions
+): Promise<IHubFeedback> {
+  let getPrms;
+  if (isGuid(identifier)) {
+    // get item by id
+    getPrms = getItem(identifier, requestOptions);
+  } else {
+    getPrms = getItemBySlug(identifier, requestOptions);
+  }
+  return getPrms.then((item) => {
+    if (!item) return null;
+    return convertItemToFeedback(item, requestOptions);
+  });
+}
+
+/**
+ * @private
+ * Convert a Hub Feedback Item into a Hub Feedback entity, fetching any additional
+ * information that may be required
+ * @param item the feedback item
+ * @param auth request options
+ * @returns a promise that resolves a IHubFeedback object
+ */
+export async function convertItemToFeedback(
+  item: IItem,
+  requestOptions: IHubRequestOptions
+): Promise<IHubFeedback> {
+  const model = await fetchModelFromItem(item, requestOptions);
+  let entitySettings;
+  try {
+    entitySettings = await fetchSetting({ id: item.id, ...requestOptions });
+  } catch (e) {
+    const defaultSettings = getDefaultEntitySettings("feedback");
+    entitySettings = {
+      id: null,
+      ...defaultSettings,
+    };
+  }
+  model.entitySettings = entitySettings;
+  const mapper = new PropertyMapper<Partial<IHubFeedback>, IModel>(
```

Comment:
In order to make the determination if a map question is configured for the survey, we'll need to fetch the form's configuration from within this method. The form configuration can live in a couple different places depending on whether we're working with a survey created using the Survey123 web application or the Survey123 Connect desktop application. A [getFormJSON](https://github.com/ArcGIS/opendata-ui/blob/master/packages/ember-arcgis-hub-components/addon/utils/item-types/survey-utils.js#L202) utility exists that abstracts the differences of retrieving the configuration for a form created in Survey123 web app vs Survey123 Connect. We should be able to avoid hoisting the [getInfoFile](https://github.com/ArcGIS/opendata-ui/blob/master/packages/ember-arcgis-hub-components/addon/utils/item-types/survey-utils.js#L290) utility used by `getFormJSON` in lieu of using the [getItemInfo](https://github.com/Esri/arcgis-rest-js/blob/main/packages/arcgis-rest-portal/src/items/get.ts#L378) RESTJS method that was added _after_ the survey feature was created.

I envision our code in this function looking something like:

```ts
model.formJSON = await getFormJson(item.id, requestOptions);
```